src/main/java/org/elasticsearch/action/fieldstats/FieldStatsShardRequest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.fieldstats;
+
+import org.elasticsearch.action.support.broadcast.BroadcastShardOperationRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.index.shard.ShardId;
+
+import java.io.IOException;
+
+/**
+ */
+public class FieldStatsShardRequest extends BroadcastShardOperationRequest {
+
+    private String[] fields;
+
+    public FieldStatsShardRequest() {
+    }
+
+    public FieldStatsShardRequest(ShardId shardId, FieldStatsRequest request) {
+        super(shardId, request);
+        this.fields = request.fields();
+    }
+
+    public String[] fields() {
```

Comment:
@javanna FieldStatsShardRequest isn't user facing, but the FieldStatsRequest is. 

I don't like fields(String...), but your point is valid, so I'll make sure that FieldStatsRequest has this method.

At some point we should change the current getter/setter pattern on all request / response classes to the standard getter/setter pattern...